push.bash

```diff
@@ -33,4 +34,12 @@ if [ -z "$BUF_COMMAND" ]; then
   fail "$NOT_INSTALLED_MESSAGE"
 fi
 
+if [ "$BUF_TRACK" != "main" ]; then
+  "${BUF_COMMAND}" push --track "${BUF_TRACK}" --help >/dev/null 2>&1 ||
+    fail "The installed version of buf does not support setting the track. Please use buf v1.0.0-rc11 or newer."
```

Comment:
I'd prefer testing the version of `${BUF_COMMAND}` to this, as we'll print this message for unrelated failures, which is confusing.

Reply:
I updated it to only print this message if the output from `buf push --track example --help` includes "unknown flag: --track"